backend/pkg/log/server/log_service_server_impl.go

```diff
@@ -42,15 +49,25 @@ func (lss *LogServiceServerImpl) Export(
 			serviceName := scopeLog.Scope.Name
 			for _, log := range scopeLog.LogRecords {
 				typedLog := typeLog(log, serviceName)
-				err := lss.cache.Put(typedLog.Service, []model.LogEntry{typedLog})
-				logWithClusterId, err := lss.logProcessor.ParseLogWithMessage(serviceName, typedLog, ctx)
-				if err != nil {
-					lss.logger.Error("Failed to parse log with message", zap.Error(err))
-				}
-				err = lss.cache.Put(logWithClusterId.ClusterId, []model.LogEntry{logWithClusterId})
-				if err != nil {
-					lss.logger.Error("Failed to put log in cache", zap.Error(err))
-				}
+
+				go func(typedLog model.LogEntry, serviceName string) {
```

Comment:
Added a goroutine here because the gRPC context would be cancelled as we're taking way too long to service the request.